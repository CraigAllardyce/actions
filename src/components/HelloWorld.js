import React, { useState } from 'react';
import { css } from '@emotion/react';

export default function HelloWorld() {
    const [colour, setColour] = useState('#1dbd45');

    const setBlue = () => setColour('#0fc8f2');

    const setGreen = () => setColour('#1dbd45');

    const setYellow = () => setColour('#ffbd53');

    const setRed = () => setColour('#ef4023');

    return (
        <div
            css={css`
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;
                height: calc(100vh - 120px);
            `}
        >
            <h1 data-test="page-title" style={{ color: colour }}>
                Hello World - 24
            </h1>

            <div
                css={css`
                    .btn {
                        margin: 0 8px;
                    }
                `}
            >
                <button type="button" className="btn btn-neutral" onClick={setBlue}>
                    Blue
                </button>

                <button type="button" className="btn btn-neutral" onClick={setGreen}>
                    Green
                </button>

                <button type="button" className="btn btn-neutral" onClick={setYellow}>
                    Yellow
                </button>

                <button type="button" className="btn btn-neutral" onClick={setRed}>
                    Red
                </button>
            </div>
        </div>
    );
}
