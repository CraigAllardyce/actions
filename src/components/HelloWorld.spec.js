import React from 'react';
import { render, configure, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import HelloWorld from './HelloWorld';

configure({ testIdAttribute: 'data-test' });

describe('render', () => {
    it('renders the component', () => {
        const { container } = render(<HelloWorld />);
        expect(container).toMatchSnapshot();
    });

    it('sets the title to blue', () => {
        const { getByText, getByTestId } = render(<HelloWorld />);
        fireEvent.click(getByText('Blue'));
        expect(getByTestId('page-title')).toHaveStyle('color: rgb(15, 200, 242)');
    });
});

describe('addition', () => {
    describe('positive numbers', () => {
        it('should add up', () => {
            expect(1 + 2).toBe(3);
        });
    });
});
