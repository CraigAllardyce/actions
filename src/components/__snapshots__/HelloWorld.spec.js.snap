// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders the component 1`] = `
<div>
  <div
    class="css-1kntp3v-HelloWorld"
  >
    <h1
      data-test="page-title"
      style="color: rgb(29, 189, 69);"
    >
      Hello World - 24
    </h1>
    <div
      class="css-absu32-HelloWorld"
    >
      <button
        class="btn btn-neutral"
        type="button"
      >
        Blue
      </button>
      <button
        class="btn btn-neutral"
        type="button"
      >
        Green
      </button>
      <button
        class="btn btn-neutral"
        type="button"
      >
        Yellow
      </button>
      <button
        class="btn btn-neutral"
        type="button"
      >
        Red
      </button>
    </div>
  </div>
</div>
`;
